services/core-api/app/api/mines/permits/permit_amendment/models/permit_amendment.py

```diff
@@ -160,14 +165,12 @@ def validate_received_date(self, key, received_date):
 
     @validates('issue_date')
     def validate_issue_date(self, key, issue_date):
-        if self.permit_amendment_type_code != 'OGP':
-            original_permit_amendment = self.query.filter_by(permit_id=self.permit_id).filter_by(
-                permit_amendment_type_code='OGP').first()
-            if original_permit_amendment and original_permit_amendment.issue_date:
-                if issue_date and original_permit_amendment.issue_date > issue_date.date():
-                    raise AssertionError(
-                        'Permit amendment issue date cannot be before the permits First Issued date.'
-                    )
+        # TODO DO NOT REMOVE NEXT LINE. If this validation removed then exception will be thrown on permit creation/editing:
```

Comment:
Is this comment actually driving some sort of validation?

Reply:
For some reason, if you remove db querying in this method it causes sqlalchemy to randomly fail with strange errors